### PR TITLE
Fixing serialization issue in java

### PR DESCRIPTION
**Pull request by vinagesh**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java [hunk 1]
```diff
@@ -70,11 +77,6 @@ public static DigitalTwinAsyncClient createFromConnectionString(String connectio
      */
     public <T> Observable<T> getDigitalTwin (String digitalTwinId, Class<T> clazz)
     {
-        if(clazz == null)
```
Comment (vinagesh) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:73]: This logic is only needed in the max overload to make testing more efficient. I've cleaned up duplicate logic.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java [hunk 2]
```diff
@@ -40,12 +43,16 @@
         ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
         SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
         String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+        final SimpleModule stringModule = new SimpleModule("String Serializer");
+        stringModule.addSerializer(new DigitalTwinStringSerializer(String.class, objectMapper));
```
Comment (bikamani) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:47]: To understand, why do we need to do this?

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientComponentTests.java [hunk 3]
```diff
@@ -42,7 +39,7 @@
 @DigitalTwinTest
```
Comment (barustum) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientComponentTests.java:39]: do you need to update this to DigitalTwinClientTest?

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java (hunk 2 above)
Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:47]: Will sync offline with you on this.

**Review by bikamani — APPROVED** [review]

**Pull request merged**